src/ui/display.c

```diff
@@ -479,6 +482,15 @@ UX_FLOW(ux_display_unverified_segwit_inputs_flow,
         &ux_display_continue_step,
         &ux_display_reject_step);
 
+// FLOW to warn about segwitv1 inputs with non-default sighash
+// #1 screen: warning icon + "Non default sighash"
+// #2 screen: "continue" button
+// #3 screen: "reject" button
+UX_FLOW(ux_display_nondefault_sighash_flow,
+        &ux_nondefault_sighash_flow_1_step,
+        &ux_display_continue_step,
+        &ux_display_reject_step);
+
```

Comment:
Maybe we can be a bit more aggressive and show a `ux_display_reject_if_not_sure_step` ("Reject if you're not sure") screen, until we figure out how people will want to use it and how we do it securely.

It is probably necessary to show more info to safely sign in some protocols (e.g. what inputs have different sighash, what is the sighash), but we can figure this out later − the scary warning should suffice for now.